modules/fundamental/src/source_document/model/mod.rs

```diff
@@ -56,3 +65,34 @@ impl TryInto<StorageKey> for SourceDocument {
         (&self).try_into()
     }
 }
+
+#[async_trait::async_trait]
+pub trait DeleteSourceDocument {
+    async fn delete_source_document(&self, db: &DatabaseTransaction) -> Result<(), DbErr>;
+}
+
+async fn delete_source_document_by_id(
+    db: &DatabaseTransaction,
+    source_document: Option<SourceDocument>,
```

Comment:
Wouldn't `Option<&SourceDocument>` be enough. Reducing the `clone`?

Reply:
Yup. Fixed!